src/parser/binary-operation-parser.ts

```diff
@@ -64,108 +57,98 @@ export type HavingGrouper<DB, TB extends keyof DB> = (
+  operator: BinaryOperatorExpression,
+  rightOperand: ReferenceExpression<any, any>
+): BinaryOperationNode {
+  return BinaryOperationNode.create(
+    parseReferenceExpression(leftOperand),
+    parseOperator(operator),
+    parseReferenceExpression(rightOperand)
   )
 }
 
-export function parseExistFilter(arg: ExistsExpression<any, any>): FilterNode {
-  return parseExistExpression('exists', arg)
-}
-
-export function parseNotExistFilter(
-  arg: ExistsExpression<any, any>
-): FilterNode {
-  return parseExistExpression('not exists', arg)
-}
-
-export function parseFilter(type: FilterType, args: any[]): OperationNode {
+export function parseFilterExpression(
+  type: FilterExpressionType,
+  args: any[]
+): OperationNode {
   if (args.length === 3) {
-    return parseThreeArgFilter(args[0], args[1], args[2])
-  } else if (args.length === 1) {
-    return parseOneArgFilter(type, args[0])
+    return parseBinaryOperation(args[0], args[1], args[2])
+  }
+
+  if (args.length === 1) {
+    return parseOneArgFilterExpression(type, args[0])
   }
 
-  throw createFilterError(type, args)
+  throw createFilterExpressionError(type, args)
 }
 
-function parseThreeArgFilter(
-  left: ReferenceExpression<any, any>,
-  op: FilterOperator,
-  right: FilterValueExpressionOrList<any, any, any>
-): FilterNode {
-  if ((op === 'is' || op === 'is not') && (isNull(right) || isBoolean(right))) {
-    return parseIsFilter(left, op, right)
+function parseBinaryOperation(
+  leftOperand: ReferenceExpression<any, any>,
+  operator: BinaryOperatorExpression,
+  rightOperand: OperandValueExpressionOrList<any, any, any>
+): BinaryOperationNode {
+  if (
+    (operator === 'is' || operator === 'is not') &&
+    (isNull(rightOperand) || isBoolean(rightOperand))
+  ) {
+    return parseIs(leftOperand, operator, rightOperand)
   }
 
-  return FilterNode.create(
-    parseReferenceExpression(left),
-    parseFilterOperator(op),
-    parseValueExpressionOrList(right)
+  return BinaryOperationNode.create(
+    parseReferenceExpression(leftOperand),
+    parseOperator(operator),
+    parseValueExpressionOrList(rightOperand)
   )
 }
 
-function parseIsFilter(
-  left: ReferenceExpression<any, any>,
-  op: 'is' | 'is not',
-  right: null | boolean
+function parseIs(
+  leftOperand: ReferenceExpression<any, any>,
+  operator: 'is' | 'is not',
+  rightOperand: null | boolean
 ) {
-  return FilterNode.create(
-    parseReferenceExpression(left),
-    parseFilterOperator(op),
-    ValueNode.createImmediate(right)
+  return BinaryOperationNode.create(
+    parseReferenceExpression(leftOperand),
+    parseOperator(operator),
+    ValueNode.createImmediate(rightOperand)
   )
 }
 
-function parseFilterOperator(op: FilterOperator): OperationNode {
-  if (isString(op) && OPERATORS.includes(op)) {
-    return OperatorNode.create(op)
-  } else if (isOperationNodeSource(op)) {
-    return op.toOperationNode()
+function parseOperator(operator: BinaryOperatorExpression): OperationNode {
```

Comment:
We still need to have a distinction between comparison operators and operators in general. We should only accept comparison operators in `where`, `having` and `on`.

Reply:
Agreed! That was the plan, the naming wasn't accurate. 😢